integration_tests/suite/test_mobile_callback.py

```diff
@@ -914,6 +1060,31 @@ def test_incoming_call_workflow(self):
             ),
         )
 
+        def assert_incoming_call_notification(request):
+            assert_that(
+                request,
+                has_entry(
+                    'message',
+                    has_entry(
+                        'data',
+                        has_entries(
+                            notification_type='incomingCall', items=instance_of(str)
+                        ),
+                    ),
+                ),
+            )
+            data = json.loads(request['message']['data']['items'])
+            assert_that(
+                data,
+                has_entries(
+                    peer_caller_id_number='caller-id',
+                    creation_time=call_creation_time.isoformat(timespec='seconds'),
+                    notification_time=a_timestamp(),
+                ),
+            )
+
+        self.assert_last_fcm_request(assert_incoming_call_notification)
```

Comment:
Suggestion: this could be turned into a context manager to make it slightly easier to read / use.